server/etcdserver/server.go

```diff
@@ -1545,9 +1546,20 @@ func (s *EtcdServer) isLearnerReady(id uint64) error {
 		return membership.ErrIDNotFound
 	}
 
+	var learnerReadyPercent float64
 	leaderMatch := rs.Progress[leaderID].Match
+
+	if leaderMatch > 0 {
```

Comment:
Why do you add this new condition? It will be much easier to review the PR if we limit the change to just adding a log.

Reply:
I thought it will be better to have a safety for `leaderMatch`
```go 
learnerReadyPercent = float64(learnerMatch) / float64(leaderMatch)
```
If doesn't make sense then I can revert it back.